src/restconfig/src/test/java/org/geoserver/rest/resources/ResourceTest.java

```diff
@@ -0,0 +1,310 @@
+        Map<String, String> mapHTML = new HashMap<String, String>();        
+        mapHTML.put("x", "http://www.w3.org/1999/xhtml");
+        NS_HTML = new SimpleNamespaceContext(mapHTML);        
+    }          
+    
+    @Before
+    public void initialise() throws IOException {                
+        myRes = getDataDirectory().get("/mydir/myres");
+        try (OutputStreamWriter os = new OutputStreamWriter(myRes.out())) {
+            os.append("This is my test.");
+        }
+        
+        try (OutputStreamWriter os = new OutputStreamWriter(getDataDirectory().get("/mydir2/fake.png").out())) {
+            os.append("This is not a real png file.");
+        }
+
+        IOUtils.copyStream(getClass().getResourceAsStream("testimage.png"),
+                getDataDirectory().get("/mydir2/imagewithoutextension").out(), true, true);
+    }
+    
+    @Test
+    public void testResource() throws Exception {
+        String str = getAsString("/rest/resource/mydir/myres");
+        Assert.assertEquals("This is my test.\n", str);
+    }
+    
+    @Test
+    public void testResourceMetadataXML() throws Exception {
+        XMLUnit.setXpathNamespaceContext(NS_XML);
+        Document doc = getAsDOM("/rest/resource/mydir/myres?operation=metadata&format=xml");
+        //print(doc);
+        XMLAssert.assertXpathEvaluatesTo("myres", "/ResourceMetadata/name", doc);
+        XMLAssert.assertXpathEvaluatesTo("/mydir", "/ResourceMetadata/parent/path", doc);
+        XMLAssert.assertXpathEvaluatesTo("http://localhost:8080/geoserver/rest/resources/mydir?format=xml", 
+                "/ResourceMetadata/parent/atom:link/@href", doc);
+        XMLAssert.assertXpathEvaluatesTo(FORMAT.format(myRes.lastmodified()),
+                "/ResourceMetadata/lastModified", doc);
+    }
+    
+    @Test
+    public void testResourceMetadataJSON() throws Exception {
+        JSON json = getAsJSON("/rest/resource/mydir/myres?operation=metadata&format=json");
+        //print(json);
+        String expected = "{\"ResourceMetadata\": {"
+                + "  \"name\": \"myres\","
+                + "  \"parent\":   {"
+                + "    \"path\": \"/mydir\","
+                + "    \"link\": {"
+                + "       \"href\": \"http://localhost:8080/geoserver/rest/resources/mydir?format=json\","
+                + "       \"rel\": \"alternate\",                "
+                + "       \"type\": \"application/json\""
+                + "     }"
+                + "   },"
+                + "  \"lastModified\": \"" + FORMAT.format(myRes.lastmodified()) + "\","
+                + "  \"type\": \"resource\""
+                + "}}";
+        JSONAssert.assertEquals(expected, (JSONObject) json);
+    }
+    
+    @Test
+    public void testResourceMetadataHTML() throws Exception {
+        XMLUnit.setXpathNamespaceContext(NS_HTML);
+        Document doc = getAsDOM("/rest/resource/mydir/myres?operation=metadata&format=html");
+        print(doc);
```

Comment:
@NielsCharlier please do not print to stdout during tests. I see that you commented out the print lines in other tests, but you missed this one.  :-)